src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -188,7 +190,84 @@ export const checkNewBoundaryEnabled = async (
 }
 
 /**
- * !! Do NOT enable `encryption-boundary-shift-virus-scanner` feature flag until this has been completely implemented. !!
+ * Helper function to trigger virus scanning and download clean file.
+ * @param response quarantined attachment response from storage submissions v2.1+.
+ * @returns modified response with content replaced with clean file buffer and answer replaced with filename.
+ */
+const triggerVirusScanThenDownloadCleanFileChain = (
+  response: ParsedClearAttachmentResponse,
+): ResultAsync<
+  ParsedClearAttachmentResponse,
+  VirusScanFailedError | DownloadCleanFileFailedError
```

Comment:
/Suggestion: For VirusScanFailedError, it may be useful to provide the exact attachment which failed. Maybe can add something like this at the end

```
    .mapErr((error) => {
      if (error instanceof VirusScanFailedError) {
        return new VirusScanFailedError(
          `Your file ${response.filename} has failed virus scanning.`,
        )
      }
      return error
    })
```

Reply:
Yea... Okay let's do that! 

Also I realised that there's no difference in error between the lambda failing and the virus scanning detecting a malicious file 😓 Good this you flagged this haha thanks! If not, we might have a slew of 500s again :')

Reply:
Done in [f16d056](https://github.com/opengovsg/FormSG/pull/6767/commits/f16d056d2a4f8d56852d143112e463b6d3099b1f)! The bad scans throwing 500s issue is fixed in [3c86b55](https://github.com/opengovsg/FormSG/pull/6767/commits/3c86b55e4c1cef14dbef6d71b5e8860034cff477) too.